src/main/java/dev/morling/onebrc/CalculateAverage_jonathanaotearoa.java

```diff
@@ -0,0 +1,459 @@
+                return word >>> (bytesToDiscard << 3);
+            }
+            return UNSAFE.getLong(address);
+        }
+    }
+
+    private static class StationData implements Comparable<StationData> {
+
+        private final String name;
+        public final int nameHash;
+        private int tempMin;
+        private int tempMax;
+        private long tempSum;
+        private int count;
+
+        public StationData(final String name, final int nameHash, final int temp) {
+            this.name = name;
+            this.nameHash = nameHash;
+            tempMin = tempMax = temp;
+            tempSum = temp;
+            count = 1;
+        }
+
+        public void addTemp(final int temp) {
+            tempMin = min(temp, tempMin);
+            tempMax = max(temp, tempMax);
+            tempSum += temp;
+            count++;
+        }
+
+        public StationData merge(final StationData other) {
+            tempMin = min(tempMin, other.tempMin);
+            tempMax = max(tempMax, other.tempMax);
+            tempSum += other.tempSum;
+            count += other.count;
+            return this;
+        }
+
+        @Override
+        public int compareTo(final StationData other) {
+            return name.compareTo(other.name);
+        }
+
+        @Override
+        public String toString() {
+            // Only include the temp information as we're relying on TreeMap's toString for the rest.
+            return round(tempMin) + "/" + round((1.0 * tempSum) / count) + "/" + round(tempMax);
+        }
+
+        private static double round(double value) {
+            return Math.round(value) / 10.0;
+        }
+    }
+
+    /**
+     * Open addressing, linear probing, hash map repository.
+     */
+    private static class Repository {
+
+        // First prime number greater than unique keys * 10.
+        private static final int CAPACITY = 100_003;
+
+        private final StationData[] table;
+
+        public Repository() {
+            this.table = new StationData[CAPACITY];
+        }
+
+        public void put(final long nameAddress, final byte nameSize, final int nameHash, int temp) {
+            final int index = findIndex(nameHash);
+            if (table[index] == null) {
+                final byte[] nameBytes = new byte[nameSize];
+                UNSAFE.copyMemory(null, nameAddress, nameBytes, UNSAFE.arrayBaseOffset(nameBytes.getClass()), nameSize);
+                final String name = new String(nameBytes, StandardCharsets.UTF_8);
+                table[index] = new StationData(name, nameHash, temp);
+            }
+            else {
+                table[index].addTemp(temp);
+            }
+        }
+
+        public Stream<StationData> entries() {
+            return Arrays.stream(table).filter(Objects::nonNull);
+        }
+
+        private int findIndex(int nameHash) {
+            int i = (nameHash & 0x7FFFFFFF) % table.length;
+            while (table[i] != null && table[i].nameHash != nameHash) {
```

Comment:
I don't think comparign the hash is enough, you'd have to compare the actual name.

Reply:
Good point! A bit of an oversight on my behalf ;)